### ISSUE TITLE
The optimizer of clients is created every epoch?

### ISSUE THREAD

**Issue by tengerye** [AshwinRJ/Federated-Learning-PyTorch#5]
Hi, thanks for the code.

According to the [lines](https://github.com/AshwinRJ/Federated-Learning-PyTorch/blob/92526c66a44c870530ea5c5c7f7db15448cb0ff3/src/update.py#L60-L65) in `update.py`:
```
if self.args.optimizer == 'sgd':
    optimizer = torch.optim.SGD(model.parameters(), lr=self.args.lr,
                                momentum=0.5)
elif self.args.optimizer == 'adam':
    optimizer = torch.optim.Adam(model.parameters(), lr=self.args.lr,
                                 weight_decay=1e-4)
```

The optimizer is created for every epoch, is that correct?

**Comment by AshwinRJ** [comment]
Yes, it is created during every global iteration.

**Comment by tengerye** [comment]
@AshwinRJ Thank you for your kind reply. May I ask the reason? It is counter-intuitive since the learning rate of optimizer is reset during the training.

**Comment by zdhNarsil** [comment]
I also have the same question. How can you set lr decay in this setting?
